tests/common/util.rs

```diff
@@ -3634,7 +3664,88 @@ mod tests {
             .succeeds();
         std::assert_eq!(
             String::from_utf8_lossy(out.stdout()),
-            "stdin is atty\r\nstdout is atty\r\nstderr is atty\r\nterminal size: 30 80\r\n"
+            "stdin is atty\r\nterminal size: 30 80\r\nstdout is atty\r\nstderr is atty\r\n"
+        );
+        std::assert_eq!(
+            String::from_utf8_lossy(out.stderr()),
+            "This is an error message.\r\n"
+        );
+    }
+
+    #[cfg(unix)]
+    #[cfg(feature = "env")]
+    #[test]
+    fn test_simulation_of_terminal_for_stdin_only() {
+        let scene = TestScenario::new("util");
+
+        let out = scene
+            .ccmd("env")
+            .arg("sh")
+            .arg("is_atty.sh")
+            .terminal_sim_stdio(TerminalSimulation {
+                stdin: true,
+                stdout: false,
+                stderr: false,
+                ..Default::default()
+            })
+            .succeeds();
+        std::assert_eq!(
+            String::from_utf8_lossy(out.stdout()),
+            "stdin is atty\nterminal size: 30 80\nstdout is not atty\nstderr is not atty\n"
```

Comment:
Typo: `is a tty`, here and in many other places.

Fun fact: The term `tty` has a very long history: https://en.wikipedia.org/wiki/Teleprinter
The term `atty` usually refers to attorneys before court, and has no particular meaning in programming. :)

More points of evidence:
- `ENOTTY` exists, but `ENOATTY` doesn't exist.
- `ttyname(3)` is not called `attyname`: https://linux.die.net/man/3/ttyname

Reply:
thanks for providing evidence. I adapted it accordingly.
by the way, this PR is actually not fixing these issues of #6121.
The PR #6124 is fixing this. But I'm paused it till the windows terminal simulation is working.